v3/lints/etsi/lint_qcstatem_etsi_national_scheme.go

```diff
@@ -0,0 +1,144 @@
+6) Two characters according to local definition within the specified country and name registration authority,
+identifying a national scheme that is considered appropriate for national and European level, followed by the
+character ":" (colon).
+
+5.1.4 Legal person semantics identifier
+The semantics of id-etsi-qcs-SemanticsId-Legal shall be as follows.
+When the legal person semantics identifier is included, any present organizationIdentifier attribute in the subject
+field shall contain information using the following structure in the presented order:
+• 3 character legal person identity type reference;
+• 2 character ISO 3166 [2] country code;
+• hyphen-minus "-" (0x2D (ASCII), U+002D (UTF-8)); and
+• identifier (according to country and identity type reference).
+The three initial characters shall have one of the following defined values:
+1) "VAT" for identification based on a national value added tax identification number.
+2) "NTR" for identification based on an identifier from a national trade register.
+3) "PSD" for identification based on national authorization number of a payment service provider under
+Payments Services Directive (EU) 2015/2366 [i.13]. This shall use the extended structure as defined in ETSI
+TS 119 495 [3], clause 5.2.1.
+4) "LEI" for a global Legal Entity Identifier as specified in ISO 17442 [4]. The 2 character ISO 3166 [2] country
+code shall be set to 'XG'.
+5) Two characters according to local definition within the specified country and name registration authority,
+identifying a national scheme that is considered appropriate for national and European level, followed by the
+character ":" (colon)
+*************************************************************************/
+
+func init() {
+	lint.RegisterCertificateLint(&lint.CertificateLint{
+		LintMetadata: lint.LintMetadata{
+			Name:          "e_qcstatem_correct_national_scheme",
+			Description:   "This lint checks that the national scheme is well-formed when used in the serialNumber or organizationIdentifier attribute in the subject field, provided that either the natural person semantics identifier or the legal person semantics identifier is present.",
+			Citation:      "ETSI EN 319 412-1 V1.4.1, 5.1.3 Natural person semantics identifier and 5.1.4 Legal person semantics identifier",
+			Source:        lint.EtsiEsi,
+			EffectiveDate: util.ETSI_EN_319_412_1_V1_4_1_DATE,
+		},
+		Lint: NewQcStatemNationalScheme,
+	})
+}
+
+func NewQcStatemNationalScheme() lint.LintInterface {
+	return &qcStatemNationalScheme{}
+}
+
+func (l *qcStatemNationalScheme) CheckApplies(c *x509.Certificate) bool {
+	_, isPresent := util.IsQcStatemPresent(c, &util.IdQcsPkixQCSyntaxV2)
+
+	if !isPresent {
+		return false
+	}
+
+	qcs2Generic := util.ParseQcStatem(util.GetQcStatemExtValue(c), util.IdQcsPkixQCSyntaxV2)
+
+	qcs2 := qcs2Generic.(util.DecodedQcS2)
+	semanticsId := qcs2.Decoded.SemanticsId
+	re := regexp.MustCompile(`^.{2}:`)
```

Comment:
```suggestion
	re := regexp.MustCompile(`^[a-zA-Z]{2}:`)
```

I am not 100% certain of this, but do at least consider making this a bit more precise (if possible).

Reply:
sounds good. Changed.